internal/eval_runtime_sidecar/proxy/model_proxy.go

```
@@ -0,0 +1,227 @@
+//  2. If resolution fails (missing file, empty value, path traversal) the proxy returns
+//     HTTP 400 to the eval container — the request is never forwarded.
+//  3. Non-ref tokens are forwarded unchanged to defaultTarget.
+func NewModelReverseProxy(defaultTarget *url.URL, client *http.Client, logger *slog.Logger, realSecretMountPath string) *httputil.ReverseProxy {
+	rp := &httputil.ReverseProxy{
+		Transport: &modelRoundTripper{
+			inner:  &roundTripperFromClient{client: client},
+			logger: logger,
+		},
+	}
+
+	rp.Director = func(req *http.Request) {
```

golangci-lint (2.12.2)

[error] SA1019: rp.Director has been deprecated since Go 1.26 and an alternative has been available since Go 1.20: Use Rewrite instead. 

(staticcheck)